blocks/editor-media-upload/index.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * WordPress dependencies
+ */
+import { select } from '@wordpress/data';
+import { mediaUpload } from '@wordpress/utils';
+
+/**
+ * Upload a media file when the file upload button is activated.
+ * Wrapper around mediaUpload() that injects the current post ID.
+ *
+ * @param {Array}    filesList    List of files.
+ * @param {Function} onFileChange Function to be called each time a file or a temporary representation of the file is available.
+ * @param {string}   allowedType  The type of media that can be uploaded.
+ */
+export default function editorMediaUpload( filesList, onFileChange, allowedType ) {
+	mediaUpload( filesList, onFileChange, allowedType, select( 'core/editor' ).getCurrentPostId() );
```

Comment:
`select( 'core/editor' ).getCurrentPostId()` might be undefined here because the `blocks`  module doesn't depend on the `editor` module. We thought for a long time that the `blocks` module can be used independently form the `editor` module but there are several places where this assumption has been proven wrong (autocompleter, shared blocks ...).

I think we're going to refactor these two modules to consolidate all components in `editor` and leave `blocks` for registration purpose only. The first step would be to extract the library of blocks to a separate module.

Not certain how to move forward for now though. Maybe we can check for the existance of the selector before calling it? and add a clarifying comment to move this to editor once the refactoring happens.

cc @gziolo @aduth @mtias another proof that this separation is not great.

Reply:
Handled in 21f40c7e and 25bf8677

Reply:
@youknowriad, I don't need more proofs. I'm already sold on the idea of having API based `blocks`, `core-blocks` and having all UI components moved from `block` to `editor` :)
I would start with moving `block/library` to `core-blocks` module and make it dependent on both `blocks` and `editor`. It might require exposing some files from `blocks` but otherwise should be an easy task.